lib/crypto/c_src/algorithms.c

```diff
@@ -44,14 +45,18 @@ void init_rsa_opts_types(ErlNifEnv* env);
 
 void init_algorithms_types(ErlNifEnv* env)
 {
+    mtx_init_curve_types =  enif_mutex_create("init_curve_types");
     init_hash_types(env);
     init_pubkey_types(env);
     init_curve_types(env);
     init_rsa_opts_types(env);
     /* ciphers and macs are initiated statically */
 }
 
-
+void cleanup_algorithms_types(ErlNifEnv* env)
```

Comment:
Warning from our internal test build since we always run with -Wmissing-prototypes in our builds:

no previous prototype for ‘cleanup_algorithms_types’ [-Wmissing-prototypes] void cleanup_algorithms_types(ErlNifEnv* env)